### DISCUSSION TITLE
Adding a list to config

### DISCUSSION THREAD

**Discussion by martin-gif** [LazyVim/LazyVim#464] [Q&A] [answered]
I have a question, which i think should be quite easy to fix, but because I just started using nvim can't figure it out by myself.
I want to use the [ultisnips](https://github.com/SirVer/ultisnips) Plugin which has a configuration "UltiSnipsSnippetDirectories". This config gets a list of Paths to look for Files. My config looks something like this
```
return{
{
    "SirVer/ultisnips",
    config = function()
      UltiSnipsSnippetDirectories = ["~/.config/nvim/UltiSnips"]
    end,
  },
}
```
I tried some different options
- ["~/.config/nvim/UltiSnips"]
- [~."/.config/nvim/UltiSnips"]
- [$HOME."/.config/nvim/UltiSnips"]

These versions give me the error:
```
<exp> expected. Lua Syntax Check. (miss-exp)
Unexpected symbol `[`. Lua Syntax Check. (unknown-symbol)
Unexpected <exp> . Lua Syntax Check. (exp-in-action)
Unexpected symbol `]`. Lua Syntax Check. (unknown-symbol) 
```

- {[~."/.config/nvim/UltiSnips"]}
- {[$HOME."/.config/nvim/UltiSnips"]}

These versions do not give an error, but the files are still not read correctly. 
Some versions should be wrong, but I believe that one should be correct and the only problem is to pass the list.

the dictionary I want to use for the files is "~/.config/nvim/UltiSnips"
i use [LazyVim version from the website](https://www.lazyvim.org/installation) on a Mac.

**Comment by folke** [comment, marked as answer]
That's not valid Lua. Would be good to read a lua guide to get more comfortable with the lua language.

Given that the name of that option is correct, you need to do the below instead:

```lua
return{
{
    "SirVer/ultisnips",
    init = function()
      vim.g.UltiSnipsSnippetDirectories = {"~/.config/nvim/UltiSnips"}
    end,
  },
}
```

What this does:

* for vim plugins (not lua), you typically need to set vim globals before loading the plugin. You can do that with the `init()` function and you can set vim globals with `vim.g`
* lists and tables use the same syntax in lua `{}`
